### PR TITLE
Update Color Studio and make use of it in Admin Color Schemes

### PR DESCRIPTION
**Pull request by frontdevde**
### Changes proposed in this Pull Request:

* update color studio dependency to 2.4.0
* update color schemes to draw on color studio for color values

#### Context:
Since https://github.com/Automattic/wp-calypso/pull/49489 Calypso is using the latest version of Color Studio which added color tweaks to the Blue and Celadon colors that are being used by the Admin Color Schemes. For Nav Unification this is an issue as it means that the colors are now out of sync between Calypso and wp-admin. To bring them back into sync and at the same time make future maintenance easier, in this PR we're updating Admin Color Schemes to use Color Studio variables instead of hardcoded values. For full context see p9Jlb4-28F-p2.

Fixes https://github.com/Automattic/wp-calypso/issues/49826

#### Does this pull request change what data or activity we track or use?

* No

### Testing instructions:

* Activate the `WordPress.com toolbar` in Jetpack Settings > Writing
* Go to /wp-admin/profile.php
* Select the `Aquatic` color scheme
* Inspect the sidebar background, it should now show in the new blue #055d9c instead of #135e96.
* Inspect the `Update Profile` button at the bottom, it should now show in the new #.008763 instead of #007e65.
* Do a sanity check on all the updated color schemes to confirm they look as expected:
- [ ] Classic Blue
- [ ] Classic Bright
- [ ] Classic Dark
- [ ] Contrast
- [ ] Nightfall
- [ ] Powder Snow
- [ ] Sakura
- [ ] Sunset

#### Proposed changelog entry for your changes:

* No changelog entry needed

### CODE REVIEW

File: projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/classic-dark/colors.scss [hunk 1]
```diff
@@ -1,11 +1,14 @@
+// make use of color studio variables for consistency across products
+@import 'node_modules/@automattic/color-studio/dist/color-variables.scss';
+
 // core variables (core _variables.scss)
```
Comment (frontdevde) [inline projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/classic-dark/colors.scss:4]: Classic Dark is an outlier as the colors are taken from wp-admin directly. So color studio colors only apply to a few variables.

**Review by mmtr — COMMENTED** [review]
Lovely work, @frontdevde! This will indeed improve the future maintenance, good thinking.

I visually tested all the changed color schemes, and they look good to me.

However, I noted some discrepancies between the value used for some variables and the comment explaining where the value comes from. I think we should double check them and make sure they are aligned.

**Comment by getdave** [comment]
@jeherve We will need to land this on in 9.5 if possible. It's more of a fix than a feature.

**Comment by mmtr** [comment]
> @jeherve We will need to land this on in 9.5 if possible. It's more of a fix than a feature.

There are some amendments that need to be done, but I'm confident @frontdevde can get them done on Monday (unfortunately I'm not familiar enough with the color we should choose in the schemes) so we can merge this before the code freeze.

File: projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/aquatic/colors.scss [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
-$menu-current-background: #f0c930; // Calypso --color-sidebar-menu-selected-background
-$menu-submenu-text: #ffffff; // Calypso --color-sidebar-submenu-text
-$menu-submenu-background: #043959; // Calypso --color-sidebar-submenu-background
-$menu-submenu-focus-text: #f0c930; // Calypso --color-sidebar-submenu-hover-text
-$menu-submenu-current-text: #ffffff; // Calypso --color-sidebar-submenu-selected-text
+$menu-current-background: $studio-yellow-20; // Calypso --color-sidebar-menu-selected-background
+$menu-submenu-text: $studio-white; // Calypso --color-sidebar-submenu-text
+$menu-submenu-background: $studio-blue-80; // Calypso --color-sidebar-submenu-background
+$menu-submenu-focus-text: $studio-yellow-20; // Calypso --color-sidebar-submenu-hover-text
+$menu-submenu-current-text: $studio-white; // Calypso --color-sidebar-submenu-selected-text
```
[the remark this answers is not in the document]
Comment (frontdevde) [reply projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/aquatic/colors.scss:25]: In this case, the value is correct. The comments were originate from the respective values on the default color scheme. In Aquatic the `--color-sidebar-submenu-selected-text` needs to be the same as `--color-sidebar-submenu-text`. I'll update the comment to reflect this.

File: projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/aquatic/colors.scss [hunk 3]
```diff
@@ -1,33 +1,36 @@
+// make use of color studio variables for consistency across products
+@import 'node_modules/@automattic/color-studio/dist/color-variables.scss';
+
 // core variables (core _variables.scss)
-$base-color: #043959; // Calypso --color-masterbar-background
-$icon-color: #c5d9ed; // Calypso --color-sidebar-gridicon-fill
-$highlight-color: #007e65; // Calypso --color-accent
-$notification-color: #09a884; // Calypso --color-masterbar-unread-dot-background
+$base-color: $studio-blue-50; // Calypso --color-masterbar-background
```
[the remark this answers is not in the document]
Comment (frontdevde) [reply projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/aquatic/colors.scss:5]: Great spot! The value was #043959 which used to be --studio-blue-80. The variable should this definitely be $studio-blue-80 here.

File: projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/classic-blue/colors.scss [hunk 4]
```diff
@@ -1,33 +1,36 @@
...
-$menu-current-background: #50575e; // Calypso --color-sidebar-menu-selected-background
-$menu-submenu-text: #ffffff; // Calypso --color-sidebar-submenu-text
-$menu-submenu-background: #135e96; // Calypso --color-sidebar-submenu-background
-$menu-submenu-focus-text: #e68b28; // Calypso --color-sidebar-submenu-hover-text
-$menu-submenu-current-text: #ffffff; // Calypso --color-sidebar-submenu-selected-text
+$menu-current-background: $studio-gray-60; // Calypso --color-sidebar-menu-selected-background
+$menu-submenu-text: $studio-white; // Calypso --color-sidebar-submenu-text
+$menu-submenu-background: $studio-blue-60; // Calypso --color-sidebar-submenu-background
+$menu-submenu-focus-text: $studio-orange-30; // Calypso --color-sidebar-submenu-hover-text
+$menu-submenu-current-text: $studio-white; // Calypso --color-sidebar-submenu-selected-text
```
[the remark this answers is not in the document]
Comment (frontdevde) [reply projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/classic-blue/colors.scss:25]: Same as above, the value is correct. The comments originated from the respective values on the default color scheme. In Classic Blue the --color-sidebar-submenu-selected-text needs to be the same as --color-sidebar-submenu-text. I'll update the comment to reflect this.

File: projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/sakura/colors.scss [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
+
 // core variables (core _variables.scss)
-$base-color: #005042; // Calypso --color-masterbar-background
-$icon-color: #8c1749; // Calypso --color-sidebar-gridicon-fill
-$highlight-color: #2271b1; // Calypso --color-accent
-$notification-color: #c9356e; // Calypso --color-masterbar-unread-dot-background
+$base-color: $studio-celadon-70; // Calypso --color-masterbar-background
+$icon-color: $studio-pink-70; // Calypso --color-sidebar-gridicon-fill
+$highlight-color: $studio-blue-50; // Calypso --color-accent
+$notification-color: $studio-pink-50; // Calypso --color-masterbar-unread-dot-background
```
[the remark this answers is not in the document]
Comment (frontdevde) [reply projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/sakura/colors.scss:8]: The previous value was `#c9356e`, which as per https://color-studio.blog/ is `$studio-pink-50`. 
I'll update the comment to reflect that the value here deviates from how the default color scheme derives its values.

[the remark this answers is not in the document]
Comment (frontdevde) [reply projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/sakura/colors.scss:8]: While the value was consistent with the previously set value, it made me suspicious that `$studio-pink-50` isn't used in the Calypso Sakura colors at all. Upon further inspection, it turns out the value Calypso uses is `#f283aa`, which correlates to `$studio-pink-20`. This matches the `--color-masterbar-unread-dot-background` in the default color scheme i.e. Classic Bright. I'll update comment and value here. Thank you!

File: projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/sunset/colors.scss [hunk 6]
```diff
@@ -1,33 +1,36 @@
...
+
 // core variables (core _variables.scss)
-$base-color: #691c1c; // Calypso --color-masterbar-background
-$icon-color: #facfd2; // Calypso --color-sidebar-gridicon-fill
-$highlight-color: #b26200; // Calypso --color-accent
-$notification-color: #c9356e; // Calypso --color-masterbar-unread-dot-background
+$base-color: $studio-red-80; // Calypso --color-masterbar-background
+$icon-color: $studio-red-5; // Calypso --color-sidebar-gridicon-fill
+$highlight-color: $studio-orange-50; // Calypso --color-accent
+$notification-color: $studio-pink-50; // Calypso --color-masterbar-unread-dot-background
```
[the remark this answers is not in the document]
Comment (frontdevde) [reply projects/plugins/jetpack/modules/masterbar/admin-color-schemes/colors/sunset/colors.scss:8]: Same as above 👍

**Comment by frontdevde** [comment]
@mmtr Great spots there. 🙌 I've updated the PR accordingly! 👍

**Review by mmtr — APPROVED** [review]
Awesome, thanks for solving the discrepancies @frontdevde.

**Pull request merged**